crypto_kem/ledacrypt23371/clean/bf_decoding.c

```diff
@@ -0,0 +1,65 @@
+#include "bf_decoding.h"
+#include "gf2x_arith_mod_xPplusOne.h"
+#include <string.h>
+
+#define ROTBYTE(a)   ( (a << 8) | (a >> (DIGIT_SIZE_b - 8)) )
+#define ROTUPC(a)   ( (a >> 8) | (a << (DIGIT_SIZE_b - 8)) )
+
+/* The second threshold is updated before decoding retrieving it from the
+ * private key field secondIterThreshold*/
+int PQCLEAN_LEDACRYPT23371_CLEAN_thresholds[2] = {B0, (V) / 2 + 1};
+
+
+int PQCLEAN_LEDACRYPT23371_CLEAN_bf_decoding(DIGIT out[], // N0 polynomials
+                                             const POSITION_T HtrPosOnes[N0][V],
+                                             DIGIT privateSyndrome[]  //  1 polynomial
+)
+{
+
+   uint8_t unsatParityChecks[N0*P];
+   DIGIT currSyndrome[NUM_DIGITS_GF2X_ELEMENT];
+   int check;
+   int iteration = 0;
+
+   do {
+      gf2x_copy(currSyndrome, privateSyndrome);
+      memset(unsatParityChecks,0x00,N0*P*sizeof(uint8_t));
+      for (int i = 0; i < N0; i++) {
+         for (int valueIdx = 0; valueIdx < P; valueIdx++) {
+            for(int HtrOneIdx = 0; HtrOneIdx < V; HtrOneIdx++) {
+               POSITION_T tmp = (HtrPosOnes[i][HtrOneIdx]+valueIdx) >= P ?
+                                (HtrPosOnes[i][HtrOneIdx]+valueIdx) -P : (HtrPosOnes[i][HtrOneIdx]+valueIdx);
```

Comment:
There are many of these ternary if statements. Is the value of `HtrPosOnes[i][HtrOneIdx]+valueIdx` or that of `tmp` security-sensitive?